Lib/fontTools/ufoLib/glifLib.py

```diff
@@ -1082,10 +1082,14 @@ def _glifTreeFromFile(aFile):
 
 def _glifTreeFromString(aString):
     data = tobytes(aString, encoding="utf-8")
-    if etree._have_lxml:
-        root = etree.fromstring(data, parser=etree.XMLParser(remove_comments=True))
-    else:
-        root = etree.fromstring(data)
+    try:
+        if etree._have_lxml:
+            root = etree.fromstring(data, parser=etree.XMLParser(remove_comments=True))
+        else:
+            root = etree.fromstring(data)
+    except Exception as etree_exception:
+        raise GlifLibError("GLIF contains invalid XML.") from etree_exception
```

Comment:
we can't also print the name of the glyph, can we?

Reply:
we probably need to catch the exception higher in the call stack to be able to print the name -- but I think it's worth it

Reply:
Yes agreed :+1: I think the ideal exception raising would:
- print the .glif path and name in fonttools
- print further the UFO and layer name in ufoLib2

This PR is intended as a first step towards that in giving us something non-lxml/xml-specific to hang on to when we see the error at a higher level; what do you think about something like this:
```python
except GlifLibError as glif_lib_error:
    raise GlifLoadError(name, glif_path) from glif_lib_error
```
in `readGlyphFromString()`/`glifLib.GlyphSet.readGlyph()`?

Reply:
yes I think so, making a new GlifLoadError type, although currently not sure whether any of the other exceptions that are thrown should be wrapped too; e.g. `readGlyph()` currently raises KeyError, GLIFFormatError, UnsupportedGLIFFormat explicitly as well